dask/array/linalg.py

```diff
@@ -328,3 +329,93 @@ def svd(a, name=None):
     dask.array.linalg.tsqr: Actual implementation with citation
     """
     return tsqr(a, name, compute_svd=True)
+
+def _dummy_identity(x):
+    # ToDo: Remove
+    return x
+
+def lu(a):
+    import scipy.linalg
+
+    if a.ndim != 2:
+        raise ValueError
+
+    vdim = len(a.chunks[0])
+    hdim = len(a.chunks[1])
+
+    token = tokenize(a)
+    name_lu = 'lu-lu-' + token
+
+    # ToDo: Remove
+    name_l_internal = 'lu-l-internal-' + token
+    name_u_internal = 'lu-u-internal-' + token
+
+    name_p = 'lu-p-' + token
+    name_l = 'lu-l-' + token
+    name_u = 'lu-u-' + token
+
+    dsk = {}
+
+    for i in range(min(vdim, hdim)):
+
+        if i == 0:
+            target = (a.name, i, i)
+        else:
+            prevs = [(np.dot, (name_l_internal, i, p),
+                              (name_u_internal, p, i)) for p in range(i)]
+            prevs = reduce(lambda x, y: (operator.add, x, y), prevs)
```

Comment:
At the moment these are all in the same task, so they'll all be loaded in at once.  If we want to do streaming work then each computation will need to be a separate key-value pair in the dictionary.